client/site-profiler/utils/trim-domain.ts

```diff
@@ -0,0 +1,7 @@
+export default function trimDomain( value: string ) {
+	if ( ! value ) {
+		return '';
+	}
+	const decodedValue = decodeURIComponent( value );
+	return decodedValue.replace( /\s/g, '' );
```

Comment:
I think what we can handle now are those URLs with URL-encoded characters. Example, if you input:
1. `exa,mple.com` -> `exa2cmple.com`
2.  `exa(mple.com` -> `exa28mple.com`
3. `exa/mple.com` -> `mple.com`
4. Etc.

Or, generally speaking, check validity and strip away the characters that are not valid before redirecting? What do you think?

```js
const test = new URL('exa,mple.com');
// Uncaught TypeError: Failed to construct 'URL': Invalid URL
```

Reply:
For now, if a user enters a URL with space, we'll show it as an invalid URL and not touch it, but if it's other special characters it'll be removed directly and redirected to the correct domain. I've also disabled the autocorrection on the input field. Please have another look and let me know what you think!

Reply:
Hi @ouikhuan, 

I have tried testing this PR and I don't see the error/warning message about the domain being an invalid one.

For both `exam ple.com`, and `exam,ple.com` the field got submitted and the domain was changed to `exam20ple.com`, `exam2cple.com` respectively. 

There is no error/warning for `exam ple.com`:

![Screenshot 2023-10-12 at 8 34 44 PM](https://github.com/Automattic/wp-calypso/assets/5733224/cbe4cf8f-ace6-4836-83d4-5c6cabd6e591)

This is the next page for `exam ple.com`:

![Screenshot 2023-10-12 at 8 35 13 PM](https://github.com/Automattic/wp-calypso/assets/5733224/90b2fe04-ac10-4374-9977-01189314937c)

Reply:
I decided to add a valid chars check here to target special characters soon and given an error validation message. This means white spaces and other special characters like `,` or `;` will also be identified as a special character here. The only exception here is `_`. Because we might having IDN urls and they can have underscores, so I'm adding it in as a valid char.